sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosClientEncryptionKeyProperties.java

```diff
@@ -0,0 +1,145 @@
+        this.clientEncryptionKey.setEncryptionAlgorithm(encryptionAlgorithm);
+        this.clientEncryptionKey.setWrappedDataEncryptionKey(wrappedDataEncryptionKey);
+        this.clientEncryptionKey.setEncryptionKeyWrapMetadata(encryptionKeyWrapMetadata);
+    }
+
+    @Beta(value = Beta.SinceVersion.V4_13_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public String getEncryptionAlgorithm() {
+        return this.clientEncryptionKey.getEncryptionAlgorithm();
+    }
+
+    @Beta(value = Beta.SinceVersion.V4_13_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public CosmosClientEncryptionKeyProperties setEncryptionAlgorithm(String encryptionAlgorithm) {
+        this.clientEncryptionKey.setEncryptionAlgorithm(encryptionAlgorithm);
+        return this;
+    }
+
+    @Beta(value = Beta.SinceVersion.V4_13_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public byte[] getWrappedDataEncryptionKey() {
+        return this.clientEncryptionKey.getWrappedDataEncryptionKey();
+    }
+
+    @Beta(value = Beta.SinceVersion.V4_13_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public CosmosClientEncryptionKeyProperties setWrappedDataEncryptionKey(byte[] wrappedDataEncryptionKey) {
+        this.clientEncryptionKey.setWrappedDataEncryptionKey(wrappedDataEncryptionKey);
+        return this;
+    }
+
+    @Beta(value = Beta.SinceVersion.V4_13_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public EncryptionKeyWrapMetadata getEncryptionKeyWrapMetadata() {
+        return this.clientEncryptionKey.getEncryptionKeyWrapMetadata();
+    }
+
+    @Beta(value = Beta.SinceVersion.V4_13_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public CosmosClientEncryptionKeyProperties setEncryptionKeyWrapMetadata(EncryptionKeyWrapMetadata encryptionKeyWrapMetadata) {
+        this.clientEncryptionKey.setEncryptionKeyWrapMetadata(encryptionKeyWrapMetadata);
+        return this;
+    }
+
+    /**
+     * Gets the name of the resource.
+     *
+     * @return the name of the resource.
+     */
+    @Beta(value = Beta.SinceVersion.V4_13_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public String getId() {
+        return this.clientEncryptionKey.getId();
+    }
+
+    /**
+     * Sets the name of the resource.
+     *
+     * @param id the name of the resource.
+     * @return the current instance of {@link CosmosContainerProperties}.
+     */
+    @Beta(value = Beta.SinceVersion.V4_13_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public CosmosClientEncryptionKeyProperties setId(String id) {
+        this.clientEncryptionKey.setId(id);
+        return this;
+    }
+
+    /**
+     * Gets the ID associated with the resource.
+     *
+     * @return the ID associated with the resource.
+     */
+    @Beta(value = Beta.SinceVersion.V4_13_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public String getResourceId() {
```

Comment:
>bl [](start = 6, length = 2)

It is standard to expose rid for the resources in v4 Java? In .net v3 they are internal.

Reply:
We need it for some resources like container (for permission usage) , however your are right we don't need public for client encryption key. Made it internal now